compiler/ghcide/src/Development/IDE/Core/Compile.hs

```diff
@@ -245,21 +245,30 @@ getModSummaryFromBuffer fp contents dflags parsed = do
         -- To avoid silent issues where something is not processed because the date
         -- has not changed, we make sure that things blow up if they depend on the
         -- date.
-    , ms_textual_imps = imports
+    , ms_textual_imps = [imp | (False, imp) <- imports]
     , ms_hspp_file    = fp
     , ms_hspp_opts    = dflags
     , ms_hspp_buf     = Just contents
 
     -- defaults:
-    , ms_hsc_src      = HsSrcFile
+    , ms_hsc_src      = sourceType
     , ms_obj_date     = Nothing
     , ms_iface_date   = Nothing
 #ifndef GHC_STABLE
     , ms_hie_date     = Nothing
 #endif
-    , ms_srcimps      = []        -- source imports are not allowed
+    , ms_srcimps      = [imp | (True, imp) <- imports]
     , ms_parsed_mod   = Nothing
     }
+    where
+      (sourceType, derivedFile) =
+          let (stem, ext) = splitExtension fp in
+          case ext of
+          ".hs"       -> (HsSrcFile, \newExt -> stem <.> newExt)
```

Comment:
This code is also used for `DAML` which uses `.daml` as the file extension. We currently track file extensions in https://github.com/digital-asset/daml/blob/master/compiler/ghcide/src/Development/IDE/Types/Options.hs#L27. We don’t have `lhs` or `hs-boot` file equivalents for DAML so it’s fine to not support DAML-specific versions of those. Haven’t thought too deeply about this but it looks like replacing the first two cases by ``ext `elem` optExtensions`` might do the trick.

Reply:
Sure - I think I'll need to thread in the list of extensions from somewhere as it's not already a parameter and presumably not in `GhcMonad`.  Let me know if I've missed something otherwise I'll do that.

Reply:
Yeah you’re right that you have to thread the options. You can use `getIdeOptions` within `Action` to get the options.

Reply:
Would you prefer to thread the whole options record or just the list of extensions?

Alternatively, I could change the logic to just look for `"-boot"`, and not care about the specific extensions. That might be more robust for the future, e.g. if the [`-x` flag](https://downloads.haskell.org/~ghc/8.8.1/docs/html/users_guide/using.html#overriding-the-default-behaviour-for-a-file) is ever supported.

Reply:
Good point, I think just looking for `-boot` might be the best option here.

Reply:
I updated the PR.